lib/puppet/provider/dsc_base_provider/dsc_base_provider.rb

```diff
@@ -306,7 +306,15 @@ def invoke_get_method(context, name_hash)
       end
 
       # Convert DateTime back to appropriate type
-      data[type_key] = Puppet::Pops::Time::Timestamp.parse(data[type_key]) if context.type.attributes[type_key][:mof_type] =~ /DateTime/i
+      if context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil?
+        begin
+          data[type_key] = Puppet::Pops::Time::Timestamp.parse(data[type_key]) if context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil?
+        rescue ArgumentError, TypeError => e
+          # Catch any failures in the parse, output them to the context and then return nil
+          context.err("Value returned for DateTime (#{data[type_key].inspect}) failed to parse: #{e}")
+          return nil
+        end
```

Comment:
I think we want the `data[type_key] = ` assignment to the begin/rescue block itself;

As is, this implementation will attempt to assign the parsed output to the key and, if that errors, return `nil` _for the entire_ resource.

Reply:
Ah, thought that was the desired result, have changed it now